src/index.js

```diff
@@ -40,7 +41,123 @@ function iterateToDoTasks() {
+  if (storedItems) {
+    todoTasks = JSON.parse(storedItems);
+    iterateToDoTasks();
+    // return storedItems;
+  }
+}
+retrieveToDoTasks();
+
+// Add event listeners to actions
+
+// const addIcon = document.getElementById('add-icon');
+const todoForm = document.getElementById('to-do-form');
+todoForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const description = todoForm.getElementById('description').value;
+  addNewTask(description);
+  saveToDoTasks();
+  todoForm.reset();
+});
+
+const listItems = document.getElementById('list-items');
+listItems.addEventListener('change', (e) => {
+  const checkbox = e.target;
+  const { index } = checkbox.parentElement.dataset;
+  todoTasks[index].completed = checkbox.checked;
+  saveToDoTasks();
+});
+
+listItems.addEventListener('input', (e) => {
+  const input = e.target;
+  const { index } = input.parentElement.dataset;
+  editTask(index, input.value);
+  saveToDoTasks();
+});
+
+// let nearestTask;
+const taskItem = document.createElement('inner-item');
+listItems.addEventListener('click', (e) => {
+  const { target } = e;
+  // nearestTask = target.closest('#inner-item');
+  if (target.classList.contains('remove-icon')) {
+    const index = todoTasks.findIndex((task) => taskItem === task.item);
+    deleteTask(index);
+    saveToDoTasks();
+  }
+});
+
+retrieveToDoTasks();
+
+// // Saving to local storage
+// function saveToDoTasks() {
+//   localStorage.setItem('todoTasks', JSON.stringify(todoTasks));
+// }
+
+// // Retrieve items from local storage
+// function retrieveToDoList() {
+//   const storedItems = localStorage.getItem('todoTasks');
+//   if (storedItems) {
+//     todoTasks = JSON.parse(storedItems);
+//     iterateToDoTasks();
+//   }
+// }
+
+// // Adding event listener to add icons
+// // const  addTaskForm = document.getElementById('to-do-form');
+// // addTaskForm.addEventListener('submit', (e) => {
+// //   e.preventDefault();
+// //   const description = addTaskForm.elements['description'].value;
+// //   addNewTask(description);
+// //   saveToDoTasks();
+// //   addTaskForm.reset();
+// // });
+
+// // Add event listeners to the list items
+// const listItems = document.getElementById('list-items');
+// listItems.addEventListener('change', (event) => {
+//   const checkbox = event.target;
+//   const index = checkbox.parentElement.dataset.index;
+//   todoTasks[index].completed = checkbox.checked;
+//   saveToDoTasks();
+// });
+
+// listItems.addEventListener('input', (event) => {
+//   const input = event.target;
+//   const index = input.parentElement.dataset.index;
+//   editTask(index, input.value);
+//   saveToDoTasks();
+// });
+
+// listItems.addEventListener('click', (event) => {
+//   const target = event.target;
+//   const item = target.closest('li');
+//   if (target.classList.contains('delete-icon')) {
+//     const index = item.dataset.index;
+//     deleteTask(index);
+//     saveToDoTasks();
+//   }
+// });
+
+// retrieveToDoList();
```

Comment:
- [ ] Please remove the commented code from the files, It's not good practice to leave commented code.